up-core-api/uprotocol/core/udiscovery/v3/udiscovery.proto

```diff
@@ -41,331 +41,300 @@ service uDiscovery {
 
+  GetUETopics(“up:/device2”)
+  Response: [], Status: code.INVALID_ARGUMENT (passed device should match local device)
+  */
 
-// Delete one or more nodes request
-message DeleteNodesRequest {
-  repeated string uris = 1;   // uProtocol formatted URI
-}
+  rpc GetuEList(GetuEListRequest) returns (GetuEListResponse){
+    option (method_id) = 4;
+  }
 
 
-// FindNodesRequest passed to FindNodes()
-message FindNodesRequest {
-  // uProtocol formatted URI for the node to search, ex. '//vcu.VIN/core.app.hartley'
-  // shall return the 'core.app.hartley' node
-  string uri = 1;
+ /*
+  Consumers shall call GetDeviceList API to fetch the list of uEs under a given domain.
+  This is an extension of GetUEList at a higher level.
+  For example: uCDS might call on uDDS to build up its local tree in case of failures
+ */
 
-  // How deep in the node tree should the results return. A value of -1 or
-  // the field is not present means all child nodes are returned, value of 0
-  // returns only the parent node, any other value is the depth of child nodes
-  optional int32 depth = 2;
-}
+ /* Example: 
+  GetDeviceList(“up:/domain”)
+  Response: [“up:/domain/device1”, “up:/domain/device2”], Status: code.OK 
 
+  GetUETopics(“up:/domain2”)
+  Response: [], Status: code.INVALID_ARGUMENT (invalid domain)
 
-// FindNodesResponse that is returned from the FindNodes() API
-message FindNodesResponse {
-  // List of node information
-  repeated Node nodes = 1;
+  */
+  rpc GetDeviceList(GetDeviceListRequest) returns (GetDeviceListResponse){
+    option (method_id) = 5;
+  }
 
-  // Return code for the rpc call
-  uprotocol.v1.UStatus status = 2;
+  /******************************* Data replication / synchronization ******/
 
-  // Time-to-live: How long (ms) the information should be live for in the database.
-  // -1 means lives forever
-  optional int32 ttl = 3;
-}
+  /*
+    UpdateData is NOT a customer facing API. This shall be used discovery 
+    instances to push an update between to core components (streamer)/ other discovery instances.
+  */
 
+  /* 
+  Example: 
+  UpdateData(“up:/domain/device/UE”)
+  Status: code.OK 
+  */
 
-// Find 1 or more properties for a given node passed to FindNodeProperties()
-message FindNodePropertiesRequest {
-  // the uri for the node in the database
-  string uri = 1;
+  rpc SyncData(UpdateDataRequest) returns (status){
+    option (method_id) = 6;
+  }
 
-  // List of 1 or more properties names to fetch from the database
-  // **NOTE:** When this is not populated with any property name, all properties are returned
-  repeated string properties = 2;
 }
 
 
-// Returned from FindNodeProperties()
-message FindNodePropertiesResponse {
-  // a list of property name/value pairs
-  map <string, PropertyValue>  properties = 1;
-
-  // Return code for the rpc call
-  uprotocol.v1.UStatus status = 2;
-}
-
+/* All response messages shall include uprotocol status field. This shall indicate whether 
```

Comment:
We are saying now in #112 that we do not want to also pass this in the reply but the RPC returns the status per commstatus (of response umessage)